### PR TITLE
Redis cluster support

### PR DESCRIPTION
**Pull request by shudipta**
xref: https://github.com/kubedb/redis/pull/74

### CODE REVIEW

File: pkg/configure-cluster/cluster.go [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+				} else {
+					break
+				}
+			}
+		}
+
+		// now just delete the nodes
+		for i := c.Cluster.MasterCnt; i < existingMasterCnt; i++ {
+			for j := 1; j < len(nodes[i]); j++ {
+				if err = c.deleteNode(pods[0][0], nodes[i][0].IP+":6379", nodes[i][j].ID); err != nil {
```
Comment (the-redback) [inline pkg/configure-cluster/cluster.go]: May be move port number to helper method

File: pkg/configure-cluster/cluster.go [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+					return err
+				}
+				time.Sleep(time.Second * 5)
```
Comment (the-redback) [inline pkg/configure-cluster/cluster.go]: Add TODO message `TODO: Find better alternative of time.slpee`

File: pkg/configure-cluster/cluster.go [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+
+Again:
+	for {
+		if nodesConf, err = c.getClusterNodes(pods[0][0], pods[0][0].Status.PodIP); err != nil {
+			return nil, err
+		}
+
+		// ensures pods[i][j] is slave of pods[i][0]
+		nodes = processNodesConf(nodesConf)
+		for _, master := range nodes {
```
Comment (the-redback) [inline pkg/configure-cluster/cluster.go]: TODO: Needs to review

**Comment by the-redback** [comment]
xref: https://github.com/kubedb/project/issues/232

**Comment by tamalsaha** [comment]
:fireworks:

**Pull request merged**